lib/recurly/client.rb

```diff
@@ -45,12 +45,28 @@ class Client
     #   sub = client.get_subscription(subscription_id: 'uuid-abcd7890')
     #
     # @param api_key [String] The private API key
-    # @param site_id [String] The site you wish to be scoped to.
-    # @param subdomain [String] Optional subdomain for the site you wish to be scoped to. Providing this makes all the `site_id` parameters optional.
-    def initialize(api_key:, site_id: nil, subdomain: nil, **options)
+    # @param logger [Logger] A logger to use. Defaults to creating a new STDOUT logger with level INFO.
+    def initialize(api_key:, site_id: nil, subdomain: nil, logger: nil, **options)
       set_site_id(site_id, subdomain)
       set_api_key(api_key)
-      set_options(options)
+
+      if logger.nil?
+        @logger = Logger.new(STDOUT).tap do |l|
+          l.level = Logger::INFO
+        end
+      else
+        @logger = logger
```

Comment:
I should probably do some basic validation of this logger. And make the `log_<level>` calls more resilient to failures.